widgets/src/components/TaskComments/TaskCommentsContainer.js

```diff
@@ -0,0 +1,208 @@
+import TaskCommentsSkeleton from 'components/TaskComments/TaskCommentsSkeleton';
+
+const styles = {
+  divider: {
+    marginTop: '10px',
+  },
+  noComments: {
+    marginTop: '10px',
+  },
+  commentContainer: {
+    marginTop: '10px',
+    maxHeight: '400px',
+    overflowY: 'auto',
+  },
+};
+
+class TaskCommentsContainer extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: false,
+      addingComment: false,
+      config: {},
+      comments: [],
+    };
+
+    this.handleError = this.handleError.bind(this);
+    this.fetchWidgetConfigs = this.fetchWidgetConfigs.bind(this);
+    this.onClickAddComment = this.onClickAddComment.bind(this);
+    this.onClickRemoveComment = this.onClickRemoveComment.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({ loading: true }, async () => {
+      const fetchedConfig = await this.fetchWidgetConfigs();
+
+      this.setState({ config: fetchedConfig }, async () => {
+        const { config } = this.state;
+        const { taskId } = this.props;
+
+        const connection = (config && config.knowledgeSource) || '';
+        const containerId = (config && config.containerId) || '';
+        const taskContainerId = `${taskId}@${containerId}`;
+
+        try {
+          const commentsResponse = await getTaskComments(connection, taskContainerId);
+          this.setState({ comments: commentsResponse.payload || [], loading: false });
+        } catch (error) {
+          this.handleError(error.message);
+        }
+      });
+    });
+  }
+
+  onClickAddComment(comment) {
+    this.setState({ addingComment: true }, async () => {
+      const { config, comments } = this.state;
+      const { taskId, onClickAddComment } = this.props;
+
+      const connection = (config && config.knowledgeSource) || '';
+      const containerId = (config && config.containerId) || '';
+      const taskContainerId = `${taskId}@${containerId}`;
+
+      try {
+        const postResponse = await postTaskComment(connection, taskContainerId, comment);
+        this.setState({ comments: [...comments, postResponse.payload], addingComment: false });
+      } catch (error) {
+        this.handleError(error.message);
+      }
+
+      onClickAddComment(comment);
+    });
+  }
+
+  async onClickRemoveComment(id) {
+    const { config, comments } = this.state;
+    const { taskId, onClickRemoveComment } = this.props;
+
+    const connection = (config && config.knowledgeSource) || '';
+    const containerId = (config && config.containerId) || '';
+    const taskContainerId = `${taskId}@${containerId}`;
+
+    try {
+      await deleteTaskComment(connection, taskContainerId, id);
+      this.setState({ comments: comments.filter(comment => comment.id !== id) });
+    } catch (error) {
+      this.handleError(error.message);
+    }
+
+    onClickRemoveComment(id);
+  }
+
+  handleError(err) {
+    const { onError } = this.props;
+    onError(err);
```

Comment:
can we also handle errors in our App using `ErrorNotification`?  we still don't know how the custom event onError will work inside an Entando App, so it would be better if we can handle errors inside our widgets.